server/manager/src/main/java/org/apache/accumulo/manager/tableOps/compact/cancel/CancelCompactions.java

```diff
@@ -54,6 +54,19 @@ public long isReady(long tid, Manager env) throws Exception {
 
   @Override
   public Repo<Manager> call(long tid, Manager environment) throws Exception {
+
+    mutateZooKeeper(tid, tableId, environment);
+    return new FinishCancelCompaction(namespaceId, tableId);
+  }
+
+  @Override
+  public void undo(long tid, Manager env) {
+    Utils.unreserveTable(env, tableId, tid, false);
+    Utils.unreserveNamespace(env, namespaceId, tid, false);
+  }
```

Comment:
I noticed that the `undo()` of CancelCompaction doesn't actually undo the mutate of ZooKeeper. From my understanding, the `undo()` of a FATE REPO is called if the operation fails. This may not be a problem for this REPO but I think there other operations that fail to actually undo anything from the operation and just call these 2 unreserve methods. I think this is something that we should probably look into.

Reply:
I don't know that you can undo some of these operations. In this case, CancelCompaction mutates the value of a znode, which is being watched by processes performing compactions. Once it's mutated, and the processes notice, there is no going back. I'm not sure what would happen if you set the znode value back to its original value.

Reply:
That is good point. Perhaps that is why nothing is done in the undo impl. I guess the undo doesn't always work for situations like this. This made me think of a recent undo failure I saw though while doing seom Random walk testing using the MutlitTable setup. https://github.com/apache/accumulo/issues/1919. It still think it would be good to revisit the FATE operations for things like this.